File: subprojects/frida-python/examples/get_frontmost_application.py
from pprint import pformat

from pygments import highlight
from pygments.formatters import Terminal256Formatter
from pygments.lexers import PythonLexer

import frida

device = frida.get_usb_device()


def trim_icon(icon):
    result = dict(icon)
    result["image"] = result["image"][0:16] + b"..."
    return result


app = device.get_frontmost_application(scope="full")
if app is not None:
    params = dict(app.parameters)
    if "icons" in params:
        params["icons"] = [trim_icon(icon) for icon in params["icons"]]
    print(f"{app.identifier}:", highlight(pformat(params), PythonLexer(), Terminal256Formatter()))
else:
    print("No frontmost application")

File: subprojects/frida-node/examples/get_frontmost_application.ts
import frida from "frida";
import { inspect } from "util";

async function main() {
    const device = await frida.getUsbDevice();
    const application = await device.getFrontmostApplication({ scope: frida.Scope.Full });
    console.log("[*] Frontmost application:", inspect(application, {
        depth: 3,
        colors: true
    }));
}

main()
    .catch(e => {
        console.error(e);
    });
